### PR TITLE
MCNP mats_from_inp robustness

### PR DESCRIPTION
**Pull request by elliottbiondo**
Previously `read_mcnp_inp` (which reads MCNP input files and returns PyNE materials) got confused when dealing with:
1. Surface cards with transformations
2. Materials cards that start with comments

This PR fixes these issues and updates the tests accordingly.

### CODE REVIEW

File: pyne/mcnp.py [hunk 1]
```diff
@@ -1389,9 +1390,8 @@ def mat_from_inp_line(filename, mat_line, densities='None'):
     # collect all material card data on one string
     line_index = 1
     line = linecache.getline(filename, mat_line + line_index)
-
     # people sometimes put comments in materials and then this loop breaks                                                                                       # so we need to keep reading if we encounter comments
-    while line[0:5] == '     ' or line[0] == 'c' or line[0] == 'C':
+    while len(line.split()) > 0 and (line[0:5] == '     ' or line[0] in 'Cc'):
```
Comment (scopatz) [inline pyne/mcnp.py]: `line[0].lower() == 'c'` is probably best.

File: pyne/mcnp.py [hunk 2]
```diff
@@ -1325,8 +1325,9 @@ def mats_from_inp(inp):
         # information is stored in a dictionary where:
         # key = material number, value = list of densities
         if len(line.split()) > 3:
-            if line.split()[0].isdigit() is True and \
-                    line.split()[1].isdigit() is True and \
+            if line.split()[0].isdigit() and \
+                    line.split()[1].isdigit() and \
+                    not line.split()[2][0].isalpha() and \
```
Comment (scopatz) [inline pyne/mcnp.py]: Maybe this test should get split up into its own function? It is now 5 lines long.

**Comment by scopatz** [comment]
Thanks for putting this in. Just a couple of points.

**Comment by elliottbiondo** [comment]
Fixed @scopatz.

**Comment by scopatz** [comment]
If only I was broken in the first place :(

**Comment by scopatz** [comment]
Looks good. Waiting on CI.

**Comment by scopatz** [comment]
Thanks @elliottbiondo!

**Pull request merged**